### PR TITLE
feat(ci): armv7 cross builds

### LINKED ISSUE #1337: Add arm32v7 builder
Create a new file called `arm.Dockerfile` that allows you to run it on arm32v7 CPUs such as the Raspberry Pi's CPU.

### PR DESCRIPTION
**Pull request by oxy**
Hopefully should cross-compile and work for armv7!

Fixes #1337.

### CODE REVIEW

**Review by jsjoeio — APPROVED** [review]
Overall, I don't see anything concerning here. Nice work 👏

File: .github/workflows/ci.yaml
Comment (jawnsy) [inline .github/workflows/ci.yaml:262]: Is it possible to install the "latest" node? The rest of the build scripts seem to bundle the latest Node 14.x, which would help us avoid security problems.

I don't have strong feelings about pinning specific versions or using the "latest" minor/patch release, but I do think we should be consistent with the versions we're using between the native vs cross-compiled builds

Comment (oxy) [reply .github/workflows/ci.yaml:262]: Unfortunately, there's no stable URL I can point to for latest node, which is why I resolved to this :(
Even in https://nodejs.org/dist/latest-v14.x/ - you need to know the latest version to have the right URL.

Comment (jawnsy) [reply .github/workflows/ci.yaml:262]: Apparently, they publish a JSON file, so we can write a script to extract the latest version using jq or whatever: https://nodejs.org/dist/index.json

it seems... unfortunate, that there doesn't seem to be an easier way to do this

anyway, shouldn't let this hold up this PR, this LGTM

File: .github/workflows/ci.yaml
Comment (jsjoeio) [reply .github/workflows/ci.yaml:257]: ^that looks a little easier to read IMO

File: ci/build/arch-override.json
Comment (jsjoeio) [inline ci/build/arch-override.json:6]: I worry about adding an extra file just because it seems like more to maintain. Also the json means we can't really put comments to explain why we have this. Any thoughts on writing a function to pass in `rpm` or `deb` and then returning values instead?

File: ci/build/build-packages.sh
Comment (jsjoeio) [inline ci/build/build-packages.sh:53]: Maybe we could add a comment above explaining why we have to override the arch? Or why we have this function?

**Review by jsjoeio — CHANGES_REQUESTED** [review]
Wondering if we can explain why we want to to go with the `arch-override.json` approach instead of providing a function and returns values based on input. 

I'd rather us do that then add more files like this json file but open to hearing why the current approach is a better approach

File: ci/build/arch-override.json
Comment (oxy) [reply ci/build/arch-override.json:6]: My reasoning to put this in a separate file is that its easier to add overrides for other architectures should we need it in the future (say powerpc/s390x) and that it also separates code (building packages) from data (what overrides to apply when)

File: ci/build/build-packages.sh
Comment (oxy) [reply ci/build/build-packages.sh:53]: Sure! I can do that - just give me a minute.

File: ci/build/arch-override.json
Comment (jsjoeio) [reply ci/build/arch-override.json:6]: Ah, okay that makes sense then :) Let's leave as is!

File: ci/build/build-packages.sh
Comment (jsjoeio) [reply ci/build/build-packages.sh:53]: Only request then is to add a comment or two explaining why we have to override the arch and  then we should approve and merge!

File: .github/workflows/ci.yaml
Comment (oxy) [reply .github/workflows/ci.yaml:257]: Oh! Didn't know I could do it this way :o
TIL

File: ci/build/build-packages.sh
Comment (oxy) [reply ci/build/build-packages.sh:57]: Done!

**Review by jsjoeio — APPROVED** [review]
ship it!

**Pull request merged**
